frontend/src/pages/SignUpPage.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { user } from '../reducers/user';
+
+// If the user already has an account, they can click on the
+// Log In button, which will change the page state to 'login'
+// and this will redirect them to the Log In Page instead
+
+// This page shows a form in which the user can fill up their username
+// and password, those values will be saved in local storage variables
+// When the "Create account" button is clicked, this will do a fetch to
+// the endpoint for creating a new User in the database. Name and password
+// are sent in the body. If successful, the response will include the user ID
+// and access token, which we now save in the global store by dispayching those
+// corresponding actions. Otherwise an error message is returned
+
+const SIGNUP_URL = 'https://project-auth-vane-axel.herokuapp.com/users';
+
+export const SignUpPage = ({ setPage }) => {
+  const dispatch = useDispatch();
+  const [name, setName] = useState('');
+  const [password, setPassword] = useState('');
+  const error = useSelector((store) => store.user.errorMessage);
+
+  // Added this function, so that if there are any persisted error messages
+  // from login page, they will not show up here on load
+  useEffect(() => {
+    dispatch(user.actions.setErrorMessage({ errorMessage: null }));
+  }, [dispatch]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ name, password })
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error('Could not create account. Please try a different username.');
+        }
+        return res.json();
+      })
+      .then((json) => {
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        dispatch(user.actions.setAccessToken({ accessToken: json.accessToken }));
+      })
+      .catch((error) => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: error.toString() }));
+      });
+  };
```

Comment:
Just wondering why this fetch is done here and not in the redux store like the others? 😊

Reply:
Because this fetch is reaching outside the app to the Server :)